infer_unet_process.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2021 Ikomia SAS
+# Contact: https://www.ikomia.com
+#
+# This file is part of the IkomiaStudio software.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import os.path
+
+from ikomia import core, dataprocess
+import copy
+from infer_unet.unet import UNet
+import torch
+from infer_unet.predict.utils_prediction import predict_mask, mask_to_image
+from PIL import Image
+from datetime import datetime
+import numpy as np
+# Your imports below
+
+
+# --------------------
+# - Class to handle the process parameters
+# - Inherits PyCore.CWorkflowTaskParam from Ikomia API
+# --------------------
+class InferUnetParam(core.CWorkflowTaskParam):
+
+    def __init__(self):
+        core.CWorkflowTaskParam.__init__(self)
+        # Place default value initialization here
+        self.modelFile = ""
+        self.img_scale = 0.5
+        self.num_channels = 3
+        self.num_classes = 4
+        self.outputFolder= ""
+
+    def setParamMap(self, param_map):
+        # Set parameters values from Ikomia application
+        # Parameters values are stored as string and accessible like a python dict
+        self.modelFile = param_map["modelFile"]
+        self.img_scale = param_map["img_scale"]
+        self.num_channels = param_map["num_channels"]
+        self.num_classes = param_map["num_classes"]
+        self.outputFolder = param_map["outputFolder"]
+        pass
+
+    def getParamMap(self):
+        # Send parameters values to Ikomia application
+        # Create the specific dict structure (string container)
+        param_map = core.ParamMap()
+        param_map["modelFile"] = self.modelFile
+        param_map["img_scale"] = self.img_scale
+        param_map["num_channels"] = self.num_channels
+        param_map["num_classes"] = self.num_classes
+        param_map["outputFolder"] = self.outputFolder
+        return param_map
+
+
+# --------------------
+# - Class which implements the process
+# - Inherits PyCore.CWorkflowTask or derived from Ikomia API
+# --------------------
+class InferUnet(dataprocess.C2dImageTask):
+
+    def __init__(self, name, param):
+        dataprocess.C2dImageTask.__init__(self, name)
```

Comment:
Semantic segmentation task should output the input image with colormap and a CSemanticSegIO output.